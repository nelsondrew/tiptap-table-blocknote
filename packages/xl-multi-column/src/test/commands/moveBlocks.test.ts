import { describe, expect, it } from "vitest";

import { setupTestEnv } from "../setupTestEnv.js";

const getEditor = setupTestEnv();

describe("Test moveBlocksUp", () => {
  it("Move into column list", () => {
    getEditor().setTextCursorPosition("paragraph-2");

    getEditor().moveBlocksUp();

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Move out of column list", () => {
    getEditor().setTextCursorPosition("column-paragraph-0");

    getEditor().moveBlocksUp();

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Move into previous column", () => {
    getEditor().setTextCursorPosition("column-paragraph-2");

    getEditor().moveBlocksUp();

    expect(getEditor().document).toMatchSnapshot();
  });
});

describe("Test moveBlocksDown", () => {
  it("Move into column list", () => {
    getEditor().setTextCursorPosition("paragraph-1");

    getEditor().moveBlocksDown();

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Move out of column list", () => {
    getEditor().setTextCursorPosition("column-paragraph-3");

    getEditor().moveBlocksDown();

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Move into next column", () => {
    getEditor().setTextCursorPosition("column-paragraph-1");

    getEditor().moveBlocksDown();

    expect(getEditor().document).toMatchSnapshot();
  });
});
