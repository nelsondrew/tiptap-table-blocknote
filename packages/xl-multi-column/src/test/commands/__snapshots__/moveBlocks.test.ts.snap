// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test moveBlocksDown > Move into column list 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Paragraph 1",
                "type": "text",
              },
            ],
            "id": "paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 2",
        "type": "text",
      },
    ],
    "id": "paragraph-2",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test moveBlocksDown > Move into next column 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 1",
        "type": "text",
      },
    ],
    "id": "paragraph-1",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 2",
        "type": "text",
      },
    ],
    "id": "paragraph-2",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test moveBlocksDown > Move out of column list 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 1",
        "type": "text",
      },
    ],
    "id": "paragraph-1",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Column Paragraph 3",
        "type": "text",
      },
    ],
    "id": "column-paragraph-3",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 2",
        "type": "text",
      },
    ],
    "id": "paragraph-2",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test moveBlocksUp > Move into column list 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 1",
        "type": "text",
      },
    ],
    "id": "paragraph-1",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Paragraph 2",
                "type": "text",
              },
            ],
            "id": "paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test moveBlocksUp > Move into previous column 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 1",
        "type": "text",
      },
    ],
    "id": "paragraph-1",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 2",
        "type": "text",
      },
    ],
    "id": "paragraph-2",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test moveBlocksUp > Move out of column list 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 1",
        "type": "text",
      },
    ],
    "id": "paragraph-1",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Column Paragraph 0",
        "type": "text",
      },
    ],
    "id": "column-paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 2",
        "type": "text",
      },
    ],
    "id": "paragraph-2",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;
